src/Common/SFA.DAS.ProviderPayments.Domain.Kernel/Domain/Extensions/TransactionTypeExtensions.cs

```diff
@@ -0,0 +1,12 @@
+﻿using SFA.DAS.Payments.DCFS.Domain;
+
+namespace SFA.DAS.ProviderPayments.Domain.Kernel.Domain.Extensions
+{
+    public static class TransactionTypeExtensions
+    {
+        public static bool ValidForTransactionTypeGroup(this TransactionType source, TransactionTypeGroup transactionTypeGroup)
+        {
+            return transactionTypeGroup.ValidTransactionTypes().Contains(source);
+        }
+    }
+}
```

Comment:
I know this is one way to solve the problem but, if we didn't use the RawEarnings class, but had transformed it into a structure which had these groupings separated, such as 
  TransactionTypeGroupOnProgLearning  {  
        TotalValue : 9999,
        Date : dd/mm/yy,
        Transactions : []
}
 TransactionTypeGroup.NinetyDayIncentives { }

would this have made the processing easier. As the data has been sorted and teh rest of the code cold just process it which out having to worry if a transaction type can exist in one or more groups.

The rawEarning view is not really a domain model, but a DTO representation of the EarningsCalc